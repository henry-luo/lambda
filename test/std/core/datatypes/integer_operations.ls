// Test: Integer Operations
// Category: core/datatypes
// Type: positive
// Expected: 1

// Test basic arithmetic operations
let a = 10
let b = 3

// Test all basic operations
let sum = a + b        // 13
let diff = a - b       // 7
let product = a * b    // 30
let quotient = a / b   // 3 (integer division)
let remainder = a % b  // 1
let power = a ** b     // 1000

// Test unary operators
let negative = -a      // -10
let positive = +a      // 10

// Test compound assignment
a += b  // a = 13
a -= b  // a = 10
a *= b  // a = 30
a /= b  // a = 10
a %= b  // a = 1
a **= b // a = 1

// Final check
1
