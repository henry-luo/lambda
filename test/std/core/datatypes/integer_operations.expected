09:15:42 [NOTICE] Start transpiling test/std/core/datatypes/integer_operations.ls...
09:15:42 [ERROR] Syntax tree has errors.
09:15:42 [ERROR] Syntax error at Ln 16, Col 16 - 16, Col 17: ERROR
09:15:42 [ERROR] ERROR node at Ln 16, Col 16 - 16, Col 17
09:15:42 [ERROR] Syntax error at Ln 23, Col 4 - 23, Col 5: ERROR
09:15:42 [ERROR] ERROR node at Ln 23, Col 4 - 23, Col 5
09:15:42 [ERROR] Syntax error at Ln 24, Col 4 - 24, Col 5: ERROR
09:15:42 [ERROR] ERROR node at Ln 24, Col 4 - 24, Col 5
09:15:42 [ERROR] Syntax error at Ln 25, Col 4 - 25, Col 5: ERROR
09:15:42 [ERROR] ERROR node at Ln 25, Col 4 - 25, Col 5
09:15:42 [ERROR] Syntax error at Ln 26, Col 4 - 26, Col 5: ERROR
09:15:42 [ERROR] ERROR node at Ln 26, Col 4 - 26, Col 5
09:15:42 [ERROR] Syntax error at Ln 27, Col 4 - 27, Col 5: ERROR
09:15:42 [ERROR] ERROR node at Ln 27, Col 4 - 27, Col 5
09:15:42 [ERROR] Syntax error at Ln 28, Col 4 - 28, Col 6: ERROR
09:15:42 [ERROR] ERROR node at Ln 28, Col 4 - 28, Col 6
09:15:42 [ERROR] Error: Failed to compile the function.
##### Script 'test/std/core/datatypes/integer_operations.ls' executed: #####
null
